### PR TITLE
Story 3: Try again functionality for match component

### PR DESCRIPTION
**Pull request by rosakng**
- Backend socket functions created by Jeffrey (used for stories 2, 3 and 4)
- added button to match page for try again functionality for host 
- hooked up button to socket events for redirecting users and host back to session page

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -54,6 +54,23 @@ io.on('connection', (socket) => {
         callback();
     });
 
+    socket.on('try_again_event', ({roomId}) => {
```
Comment (jeffreyquiambao) [inline server/index.js:57]: everything looks good! i think you can just remove this since my PR already has the code, it just has to be merged

**Review by jeffreyquiambao — APPROVED** [review]

File: server/index.js (hunk 1 above)
Comment (rosakng) [reply server/index.js:57]: sounds good! will do

File: wewatch-client/src/components/Lobby/Lobby.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const Lobby = ({location}) => {
     const [users, setUsers] = useState([]);
     const [goSwipe, setGoSwipe] = useState(false);
     const [topTenMovies, setTopTenMovies] = useState(null);
+    console.log(location);
```
Comment (dlhxu) [inline wewatch-client/src/components/Lobby/Lobby.js:16]: nitpick but I think this console.log could potentially be confusing in the future when trying to debug. Would it be possible to remove this or reformat it to make it's purpose more clear?

File: wewatch-client/src/components/Match/Match.js [hunk 3]
```diff
@@ -1,44 +1,83 @@
...
+    }
+
+    //listen to try again event
+    useEffect(() => {
+        socket.on('tryAgainRedirectHost', () => {
+            if (isHost) {
+                setTryAgainHost(true);
+            } 
+       })
+       return () => { socket.off('tryAgainRedirectHost')};
```
Comment (dlhxu) [inline wewatch-client/src/components/Match/Match.js:32]: Good catch making sure that socket.off is called here! In previous sprints, this issue caused memory leaks so ensuring we have the correct lifecycle functions being called will undoubtedly save us time debugging in the future.

File: wewatch-client/src/components/Lobby/Lobby.js (hunk 2 above)
Comment (rosakng) [reply wewatch-client/src/components/Lobby/Lobby.js:16]: Ah for sure! Actually just forgot to take this one out. Good catch!

**Review by dlhxu — APPROVED** [review]

File: wewatch-client/src/views/swiping/swiping-container.js [hunk 4]
```diff
@@ -85,15 +84,20 @@ const SwipingContainer = (props) => {
...
+          { noMatch ? 
+            <Redirect to={{
+              pathname: '/noMatch',
+              state: {isHost: isHost,
+                      roomId: roomId,
+                      name: name}
+              }}/>
+            : null }
+          <h1 style={{'text-align': "center", 'margin-top': '60px'}}>
+            You've seen all potential movies for recommendation, please wait as your the others finish swiping!
```
Comment (billxsheng) [inline wewatch-client/src/views/swiping/swiping-container.js:99]: Typo here.

**Review by billxsheng — APPROVED** [review]
Good job :)

**Review by varoongupta — APPROVED** [review]

**Pull request merged**